### PR TITLE
Adding cephfs TC 83574329

### PR DESCRIPTION
**Pull request by julpark-rh**
http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-ORAEMV

Create a client for a “test directory” from root mounted directory  using “ceph fs get-or-create”
Mount the target directory
Create a actual “test directory” in the mounted directory
Set “max_mds 2”
Find out which mds is ranked at 1
Set the ceph.dir.pin extended attribute on a directory (test directory)
Store the “request” number before IOs the ranked MD”
Run IOs
“Request” number
Unpin the directory from the MDS using “-v -1”
Check if the value is set to -1

This test is expected to fail due to random mds request 

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

File: tests/cephfs/cephfs_mds_pinning/mds_pinning_verification_on_direcotry.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                cmd=f"ceph tell mds.{mds} perf dump | jq -r '.mds | .request'",
+            )
+            request = int(request[0].strip())
+            rank_1_request_after = rank_1_request_after + int(request)
+        if rank_0_request_after > rank_0_request_prev:
+            log.info(f"rank_0_prev: {rank_0_request_prev}")
+            log.info(f"rank_0_after: {rank_0_request_after}")
+            log.info(f"rank_1_prev: {rank_1_request_prev}")
+            log.info(f"rank_1_after: {rank_1_request_after}")
+            raise CommandFailed("IOs are not pinned to rank 0")
```
Comment (sumabai) [inline tests/cephfs/cephfs_mds_pinning/mds_pinning_verification_on_direcotry.py:113]: Exception msg seems incorrect, if rank_0_request_after > rank_0_request_prev , then IOs are pinned to rank_0 right? then msg could be "IOs are pinned to rank 0"

Comment (sumabai) [inline tests/cephfs/cephfs_mds_pinning/mds_pinning_verification_on_direcotry.py:113]: The test log uploaded is failing, http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-6K02VC/Active_Avtive_mds_race_condition_0.log, we need to find if the failure is expected or not,
if expected, handle the failure and not report an exception as it was expected.
If failure was not expected, and if its product issue, a BZ ID should be tagged , as other wise , a merge in this state, will cause a test failure in CI.

**Review by sumabai — CHANGES_REQUESTED** [review]

File: suites/pacific/cephfs/tier-2_cephfs_test_mds_pinning.yaml [hunk 2]
```diff
@@ -200,3 +200,9 @@ tests:
       polarion-id: CEPH-11233
       desc: Subtree Split and Subtree merging by pinning Subtrees/directories to MDS.
       abort-on-fail: false
+  - test:
+      name: mds pinning on the directory verification using request
```
Comment (hkadam134) [inline suites/pacific/cephfs/tier-2_cephfs_test_mds_pinning.yaml:204]: Can you rename the name and description as it's not clear.
Maybe you can replace it with "map and unmap directory trees to a mds rank "

Comment (hkadam134) [reply suites/pacific/cephfs/tier-2_cephfs_test_mds_pinning.yaml:204]: rename the file name too.

File: tests/cephfs/cephfs_mds_pinning/mds_pinning_verification_on_direcotry.py (hunk 1 above)
Comment (julpark-rh) [reply tests/cephfs/cephfs_mds_pinning/mds_pinning_verification_on_direcotry.py:113]: I think BZ is not created so far, I am talking to VEnky regarding to this.. If it is a product issue, I will create a BZ and will add it.

File: suites/pacific/cephfs/tier-2_cephfs_test_mds_pinning.yaml (hunk 2 above)
Comment (julpark-rh) [reply suites/pacific/cephfs/tier-2_cephfs_test_mds_pinning.yaml:204]: I will do

File: tests/cephfs/cephfs_mds_pinning/mds_pinning_verification_on_direcotry.py (hunk 1 above)
Comment (julpark-rh) [reply tests/cephfs/cephfs_mds_pinning/mds_pinning_verification_on_direcotry.py:113]: I have updated it now

File: tests/cephfs/cephfs_mds_pinning/mds_pinning_verification_on_direcotry.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+from ceph.ceph import CommandFailed
+from tests.cephfs.cephfs_utilsV1 import FsUtils
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    try:
```
[the remark this answers is not in the document]
Comment (julpark-rh) [reply tests/cephfs/cephfs_mds_pinning/mds_pinning_verification_on_direcotry.py:15]: added

**Review by sumabai — DISMISSED** [review]
LGTM

**Comment by hkadam134** [comment]
Do you have the latest logs julius ?

**Comment by hkadam134** [comment]
@julpark-rh - Can you update the BZ number in the Script so that it helps while triaging the failures ?

**Comment by julpark-rh** [comment]
> @julpark-rh - Can you update the BZ number in the Script so that it helps while triaging the failures ?

okay I will

**Review by sumabai — APPROVED** [review]
LGTM

**Review by AmarnatReddy — APPROVED** [review]

**Pull request merged**